2-Draft-Accepted/RFCxxxx-DSC-for-PowerShell7.md

```diff
@@ -0,0 +1,117 @@
+OMI as a host on Linux, and [MOF](https://docs.microsoft.com/windows/win32/wmisdk/managed-object-format--mof-) for configuratio and schema.
+
+MOF as a format is not something most developers are familiar with and not much tooling support.
+However, developers of DSC resources written in PowerShell script still needed to learn MOF and also keep their MOF schema
+in sync with their script implementation.
+
+Today, when you execute a [DSC configuration script](https://docs.microsoft.com/powershell/scripting/dsc/configurations/configurations?view=powershell-7),
+the output is a MOF configuration file.
+In general, users do not need to read this MOF file as it's intended to be used by machines.
+However, MOF parsing creates a dependency on WMI/OMI.
+Instead, we will move towards a JSON configuration file format.
+
+Since there is no default agent to understand the JSON configuration file, it will be up to users or 3rd party agents to understand the configuration file.
+
+<TODO> example JSON configuratoin </TODO>
+
+### No more MOF schema and script resources
+
+Although DSC resources can be written as WMI/OMI providers, this requires writing C/C++ code and compiling for specific Linux distros and processor architectures.
+Writing PowerShell script DSC resources currently still requires a matching schema file making it more complex.
+Instead, the current plan is to [only support PowerShell class based DSC resources](https://github.com/PowerShell/PowerShell/issues/13731) in DSC for PowerShell 7.
+The reasoning and discussion is in the issue linked above.
+
+### Open Source
+
+The current [PSDesiredStateConfiguration](https://docs.microsoft.com/powershell/module/psdesiredstateconfiguration/?view=powershell-7) module is not Open Source.
+The intent is to clean up the code and prepare the GitHub repository to be made public as an Open Source project.
+
+### PowerShell engine changes
+
+To enable DSC for PowerShell 7 to evolve independently to PowerShell 7 itself, the code in the engine that handles the DSC configuration keyword will
+be seperated as a plugin and made to be part of the PSDesiredStateConfiguration module.
+
+## Alternate Proposals and Considerations
+
+### Supporting existing script based resources
+
+Existing script based resources (not class based) require a MOF schema file for PowerShell to understand the types and nested types.
+We had prototyped using a JSON based schema instead of the existing MOF schema file, but it still required the DSC resource author to
+create a schema file and ensure it aligns with the script implementation.
+
+It is simpler to create a PoewrShell class based resource which itself is the schema removing the need to understand and maintain a
+separate schema file.
+In addition, having a single model (class based) for authoring DSC resources will help the community in the long term by having a single
+consistent way to write DSC resources making it easier for new authors to find documentation and examples.
```

Comment:
consider linking to this article: https://dsccommunity.org/blog/class-based-dsc-resources/

Reply:
will do